control-plane/agents/src/bin/core/controller/registry.rs

```diff
@@ -138,21 +150,35 @@ impl Registry {
 
     /// Get the `CoreRegistryConfig` from etcd, if it exists, or use the default.
     /// If the mayastor_v1 config exists, then reuse it.
-    async fn get_config_or_panic<S: Store>(mut store: S) -> CoreRegistryConfig {
+    async fn get_config_or_panic<S: Store>(
+        mut store: S,
+        legacy_prefix_present: bool,
+    ) -> CoreRegistryConfig {
         let config = CoreRegistryConfig::new(NodeRegistration::Automatic);
-        match store.get_obj(&config.key()).await {
-            Ok(store_config) => store_config,
+        let mut config = match store.get_obj(&config.key()).await {
+            Ok(config) => config,
             Err(StoreError::MissingEntry { .. }) => {
-                store.put_obj(&config).await.expect(
-                    "Must be able to access the persistent store to persist configuration information",
-                );
+                store
+                    .put_obj(&config)
+                    .await
+                    .expect("Failed to persist configuration information");
                 config
-            },
-            Err(error) => panic!(
-                "Must be able to access the persistent store to load configuration information. Got error: '{error:#?}'"
-            ),
+            }
+            Err(error) => panic!("Failed to load configuration information: {error:?}"),
+        };
+
+        if legacy_prefix_present != config.mayastor_compat_v1() {
+            config.set_mayastor_compat_v1(legacy_prefix_present);
+            store
+                .put_obj(&config)
+                .await
+                .expect("Failed to persist configuration information");
+            warn!("Legacy V1 prefix present, enabling compatibility mode");
```

Comment:
This doesn't seem right, what if legacy_prefix_present is false but config.mayastor_compat_v1() is true? This will actually clear it which seems wrong?

Reply:
Fixed